assets/js/components/event/PresenterArea.tsx

```diff
@@ -24,87 +30,86 @@ const PresenterArea = ({ client, eventChannel, mode, setMode }: PresenterAreaPro
   const [isControlPanelAvailable, setIsControlPanelAvailable] = useState(false);
   const [isClientPresenting, setIsClientPresenting] = useState(false);
 
+  const onPresenterReady = () => {
+    setIsClientPresenting(true);
+
+    eventChannel &&
+      eventChannel.push("presenter_ready", {
+        email: client.email,
+      });
+  };
+
+  const getCurrentPresenter = () => {
+    return (
+      Object.values(presenters).find((presenter) => presenter.email == client.email) || {
+        name: client.name,
+        email: client.email,
+        status: "idle",
+        connectCallbacks: [],
+      }
+    );
+  };
+
+  useEffect(() => {
+    askForPermissions();
+    presenterArea[client.email] = new MediaStream();
+  }, []);
+
   useEffect(() => {
     const isClientPresenter = client.email in presenters;
+    const clientWantsToConnect =
+      !webrtcConnecting && webrtc == null && isClientPresenter && isClientPresenting;
+    const clientShouldDisconnect = webrtc != null && !isClientPresenter;
+
     if (isClientPresenter && !isClientPresenting) {
       setIsControlPanelAvailable(true);
-    } else if (!webrtcConnecting && webrtc == null && isClientPresenter && isClientPresenting) {
+    } else if (clientWantsToConnect) {
       webrtcConnecting = true;
-      connectWebrtc(eventChannel, client, presenters, setPresenters).then((value) => {
+      connectWebrtc(eventChannel, client, setPresenters).then((value) => {
         webrtc = value;
         setIsControlPanelAvailable(true);
         webrtcConnecting = false;
       });
-    } else if (webrtc != null && !isClientPresenter) {
+    } else if (webrtc != null && clientShouldDisconnect) {
       leaveWebrtc(webrtc, client, eventChannel);
```

Comment:
```suggestion
    } else if (clientShouldDisconnect) {
      leaveWebrtc(webrtc!, client, eventChannel);
```

Reply:
Can't do that unfortunately, linter **screms**.